pulser/sequence.py

```diff
@@ -299,6 +320,48 @@ def current_phase_ref(
 
         return self._phase_ref[basis][qubit].last_phase
 
+    def set_magnetic_field(
+        self, bx: float = 0.0, by: float = 0.0, bz: float = 30.0
+    ) -> None:
+        """Sets the magnetic field acting on the entire array.
+
+        The magnetic field vector is defined on the reference frame of the
+        atoms in the Register (with the z-axis coming outside of the plane).
+        Can only be defined before there are pulses added to the sequence.
+
+        Note:
+            The magnetic field only work in the "XY Mode". If not already
+            defined through the declaration of a Microwave channel, calling
+            this function will enable the "XY Mode".
+
+        Keyword Args:
+            bx (float): The magnetic field in the x direction (in Gauss).
+            by (float): The magnetic field in the y direction (in Gauss).
+            bz (float): The magnetic field in the z direction (in Gauss).
+        """
+        if not self._in_xy:
+            if self._channels:
+                raise ValueError(
+                    "The magnetic field can only be set in 'XY " "Mode'."
+                )
+            # No channels declared yet
+            self._in_xy = True
+        elif not self._empty_sequence:
+            # Not all channels are empty
+            raise ValueError(
+                "The magnetic field can only be set on an empty " "sequence."
+            )
+
+        mag_vector = (bx, by, bz)
+        if np.linalg.norm(mag_vector) == 0.0:
+            raise ValueError(
+                "The magnetic field must have a magnitude greater" " than 0."
+            )
```

Comment:
Is it worth considering imposing a magnitude greater than a tolerance instead of different from zero? Something like `1e-8`?

Reply:
Sure, but I'm not sure what that tolerance will be (or even if it will depend on the device) and I didn't want to make up a value. It's a good idea though, it would make more sense that way

Reply:
I would say we keep it like this for now and then define more specific values once we have them